staff/acct/sorry/sorry

```diff
@@ -1,156 +1,183 @@
+    os.chmod(user_dir + '/.sorry', 0o400)
+    shutil.chown(user_dir + '/.sorry', user=username, group='ocf')
+
+    print("Changing user's shell to sorry shell")
+
+    ocflib.infra.ldap.modify_ldap_entry(ocflib.account.utils.dn_for_username(username),
+                                        {'loginShell': sorry_shell, 'gidNumber': grp.getgrnam('sorry').gr_gid})
+
+    http_dir = '/services/http/users/{0}/{1}'.format(username[0], username)
+    print('Changing permissions on httpdir to 000')
+    if (os.path.isdir(http_dir)):
+        os.chmod(http_dir, 0o000)
+
+    print('Changing permissions on home directory to 500')
+    os.chmod(user_dir, 0o500)
+
+    # WTF is final system check for?
+    # log(rootstaffer, username)
+    send_log_mail(rootstaffer, username, open(reason_path, 'r').read())
+
+    user_mail = get_user_real_mail(username)
+    if user_mail:
+        send_sorry_mail(user_mail, username, open(reason_path, 'r').read())
+    kerberos_destroy()
+
+# Returns the real user who is running this script
+
+
+def real_user():
+    if 'SUDO_USER' in os.environ and os.environ['SUDO_USER'] != 'root':
+        return os.environ['SUDO_USER']
+    else:
+        print("The sorry.log is much more useful when it logs who you are rather than simply 'root'.")
+        username = input('Please enter your username: ')
+        return username
+
+
+def kerberos_credentials(staff_name):
+    if ('SORRY_KRB5CCNAME' in os.environ):
+        print('SORRY_KRB5CCNAME set in environment. Assuming this file contains current admin credentials.')
+        os.environ['KRB5CCNAME'] = os.environ['SORRY_KRB5CCNAME']
+    else:
+        print('Your are {0}.'.format(staff_name))
+        if (subprocess.call([shutil.which('kinit'), '{0}/admin'.format(staff_name)]) != 0):
+            raise OSError('Kinit failed, bailing out!')
+        else:
+            print('Kinit complete.')
+
+
+def kerberos_destroy():
+    return subprocess.call(['kdestroy'])
+
+
+def get_user_real_mail(username):
+    email = str(subprocess.run([shutil.which('ldapsearch'),
+                                'uid={0}'.format(username), 'mail'], stdout=subprocess.PIPE).stdout)
+    email = email[email.find('\\nmail') + 8:]
+    email = email[:email.find('\\n\\n')]
+    return email
+
+
+def log(root_staffer, sorried_user):
+    with open('/opt/acct/sorry.log', 'a') as log_file:
+        # Date style matters...?
+        log_file.write('{0} - {1} {2}'.format(str(datetime.datetime.now()), root_staffer, sorried_user))
+
+
+def send_log_mail(root_staffer, sorried_user, reason):
+    email_from = 'Open Computing Facility <sorry_log@ocf.berkeley.edu>'
+    email_subject = '[OCF] Account disabled'
+    email_body = '''
+This is an automated email sent by sorry script. This is the log of a recent sorry operation:
+Account "{0}" has been disabled by "{1}" for reason "{2}" on "{3}".
+'''.format(sorried_user, root_staffer, reason, str(datetime.datetime.now()))
```

Comment:
`dedent` (`from textwrap import dedent`) would be useful here so you can intent all this more, I also mention it further down. I do like the idea of logging to email like this, although I'm not sure how noisy it'd be. I doubt we sorry accounts often enough for it to be too annoying though.